easybuild/framework/easyblock.py

```diff
@@ -3385,6 +3385,27 @@ def sanity_check_cuda(self, cuda_dirs=None):
             self.log.info("Using configured subdirectories for binaries/libraries to verify CUDA device code: %s",
                           cuda_dirs)
 
+        # collect all files to consider
+        files_to_check = []
+        for dirpath in [os.path.join(self.installdir, d) for d in cuda_dirs]:
+            if os.path.exists(dirpath):
+                self.log.debug(f"Sanity checking files for CUDA device code under directory {dirpath}:")
+                for entry in os.listdir(dirpath):
+                    path = os.path.join(dirpath, entry)
+                    if os.path.isfile(path):
+                        self.log.debug("Sanity checking file {path} for CUDA device code")
+                        files_to_check.append(path)
+            else:
+                self.log.debug(f"Not sanity checking files in non-existing directory {dirpath}")
+
+        # also consider compiled Python modules as shared libraries (*.so) under lib/python*/site-packages
```

Comment:
This is new code. I'm ok with including it, but it is not covered in the tests.

Reply:
I'll take a look at enhancing the tests so this bit is covered too (but please go ahead and merge this already, this is secondary at best to getting https://github.com/easybuilders/easybuild-framework/pull/4692 merged at this point, especially if we want this included in the next EasyBuild release).

I tested this manually via `--sanity-check-only` on a couple of things, including `PyTorch`, it works as designed